src/app/Controllers/MoviesController.cs

```diff
@@ -30,64 +32,51 @@ public MoviesController(ILogger<MoviesController> logger, IDAL dal)
         /// <summary>
         /// Returns a JSON array of Movie objects
         /// </summary>
-        /// <param name="q">(optional) The term used to search by movie title (rings)</param>
-        /// <param name="genre">(optional) Movies of a genre (Action)</param>
-        /// <param name="year">(optional) Get movies by year (2005)</param>
-        /// <param name="rating">(optional) Get movies with a rating >= rating (8.5)</param>
-        /// <param name="actorId">(optional) Get movies by Actor Id (nm0000704)</param>
-        /// <param name="pageNumber">1 based page index</param>
-        /// <param name="pageSize">page size (1000 max)</param>
+        /// <param name="movieQueryParameters"></param>
         [HttpGet]
-        public async Task<IActionResult> GetMoviesAsync([FromQuery] string q = null, [FromQuery] string genre = null, [FromQuery] int year = 0, [FromQuery] double rating = 0, [FromQuery] string actorId = null, [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = Constants.DefaultPageSize)
+        public async Task<IActionResult> GetMoviesAsync([FromQuery] MovieQueryParameters movieQueryParameters)
         {
-            string method = GetMethodText(q, genre, year, rating, actorId, pageNumber, pageSize);
-
-            // validate query string parameters
-            var result = ParameterValidator.Movies(HttpContext?.Request?.Query, q, genre, year, rating, actorId, pageNumber, pageSize, method, logger);
-            if (result != null)
-            {
-                return result;
-            }
+            _ = movieQueryParameters ?? throw new ArgumentNullException(nameof(movieQueryParameters));
 
             // convert to zero based page index
-            pageNumber = pageNumber > 1 ? pageNumber - 1 : 0;
-
-            return await ResultHandler.Handle(dal.GetMoviesAsync(q, genre, year, rating, actorId, pageNumber * pageSize, pageSize), method, Constants.MoviesControllerException, logger).ConfigureAwait(false);
+            movieQueryParameters.PageNumber = movieQueryParameters.PageNumber > 1 ? movieQueryParameters.PageNumber - 1 : 0;
```

Comment:
should we encapsulate this in the class?

Reply:
This was tried and unfortunately it causes model validation to fail since the default for int = 0 which breaks the validation rule of (1-10000) for that property. Setting PageNumber after validation and model binding feels like the right place to do this. I'm happy to change it in another PR if we feel it's necessary.

Reply:
It seems like it should be encapsulated. If you forget to handle it in the calling code, it will cause errors.

Reply:
I'm open to suggestions on this one as I agree. However, in the spirit of the original issue this PR seeks to resolve, I would prefer to leave this existing code, which was acceptable previously and capture it as an enhancement for a future PR. If not, the only way I know how to make this work is to modify the getter/setter on the property. Do you have any other suggestions?

Reply:
discussed in call. code updated.